src/lib/dialog/dialog.ts

```diff
@@ -241,6 +236,15 @@ export class MdDialog {
       });
     }
 
+    // Close when escape keydown event occurs
+    RxChain.from(overlayRef.keyboardEvents())
+      .call(filter, event => event.type === 'keydown' && event.keyCode === ESCAPE)
+      .subscribe(() => {
+        if (!dialogRef.disableClose) {
```

Comment:
Might be a bit more concise to have this in `filter`, instead of checking it in the `subscribe`.